src/Scope.ts

```diff
@@ -343,8 +347,10 @@ export class Scope {
                         namespaces: {},
                         classStatements: {},
                         functionStatements: {},
-                        statements: []
+                        statements: [],
+                        symbolTable: new SymbolTable(parentSymbolTable)
                     };
+                    parentSymbolTable = namespaceLookup[lowerLoopName].symbolTable;
```

Comment:
Could you explain this a little bit? I _think_ this is setting every namespace part to the previous part. 
i.e. if you have `namespace a.b.c`, then `a.b.c`'s parent would be `a.b`, and `a.b`'s parent would be `a`. Is that right? If so, what benefit does that provide?

Reply:
Yep.. this is an error. all of these new symbol tables should be parented to the scope